### PR TITLE
Persist "show future events" in categories

### PR DESCRIPTION
**Pull request by bpedersen2**
Use the same logic as for past events to enable
(session-)persistent storage of the "show future
events" state.

### CODE REVIEW

File: indico/modules/categories/controllers/display.py [hunk 1]
```diff
@@ -382,22 +387,33 @@ def _process(self):
         return jsonify_data(flash=False, html=html)
 
 
-class RHShowPastEventsInCategory(RHDisplayCategoryBase):
-    """Set whether the past events in a category are automatically displayed or not"""
+class RHShowEventsInCategory(RHDisplayCategoryBase):
```
Comment (ThiefMaster) [inline indico/modules/categories/controllers/display.py]: I'd call this `RHShowEventsInCategoryBase`

File: indico/modules/categories/controllers/display.py [hunk 2]
```diff
@@ -382,22 +387,33 @@ def _process(self):
...
+        self._show_events(False)
 
     def _process_PUT(self):
-        self._show_past_events(True)
+        self._show_events(True)
+
+
+class RHShowFutureEventsInCategory(RHShowEventsInCategory):
+    """Set whether the past events in a category are automatically displayed or not"""
+    session_field = 'fetch_future_events_in'
```
Comment (ThiefMaster) [inline indico/modules/categories/controllers/display.py]: there should be an empty line after a class-level docstring

File: indico/modules/categories/controllers/display.py [hunk 3]
```diff
@@ -382,22 +387,33 @@ def _process(self):
...
+
+
+class RHShowPastEventsInCategory(RHShowEventsInCategory):
+    """Set whether the past events in a category are automatically displayed or not"""
+    session_field = 'fetch_past_events_in'
```
Comment (ThiefMaster) [inline indico/modules/categories/controllers/display.py]: same

**Comment by ThiefMaster** [comment]
changelog entry please (2.1 improvements), e.g. this:

```rst
- Add user setting whether to show future events or not by default in a
  category. Also keep the per-category status in the session (:issue:`3233`,
  thanks :user:`bpedersen2`)
```

File: indico/modules/categories/controllers/display.py (hunk 3 above)
Comment (bpedersen2) [reply indico/modules/categories/controllers/display.py]: done

File: indico/modules/categories/controllers/display.py (hunk 2 above)
Comment (bpedersen2) [reply indico/modules/categories/controllers/display.py]: done

File: indico/modules/categories/controllers/display.py (hunk 1 above)
Comment (bpedersen2) [reply indico/modules/categories/controllers/display.py]: done

**Pull request merged**